pkg/core/resources/model/resource.go

```diff
@@ -128,6 +128,13 @@ func Validate(resource Resource) error {
 	return nil
 }
 
+func Deprecations(resource Resource) []string {
+	if v, ok := interface{}(resource).(interface{ Deprecated() []string }); ok {
```

Comment:
`GetDeprecations()` or just `Deprecations()` might be a better name here no?

Reply:
Right, `Deprecations` sounds better